### ISSUE TITLE
options with * are ignored when there are options without *

### ISSUE THREAD

**Issue by wpalfi** [conan-io/conan#2393]
### Problem
I have a LibA with two options:
```python
    options = {"shared": [True, False], "test": [True, False]}
    default_options = "shared=False", "test=False"
```
LibB requires LibA. When I run
`conan info LibB/0.1@user/testing -o LibA:test=True -o *:shared=True 
`
the second option `*:shared=True` is ignored.

### Hack
In conans/model/options.py Options.propagate_upstream() I replaced 
`if not option_values:`
by
`if True:`
It seems to work with this modification.

Conan 1.0.3, Windows
- [x] I've read the [CONTRIBUTING guide](https://raw.githubusercontent.com/conan-io/conan/develop/.github/CONTRIBUTING.md).
- [x] I've specified the Conan version, operating system version and any tool that can be relevant.
- [x] I've explained the steps to reproduce the error or the motivation/use case of the question/suggestion.

**Comment by lasote** [comment]
Thanks @wpalfi! we will take a look to the bug to fix it in the next minor if possible.

**Comment by memsharded** [comment]
Fixed in 1.0.4, please update and report.
